src/llm_vm/onsite_llm.py

```diff
@@ -508,8 +508,23 @@ def __init__(self, model_uri=None, model_kwargs={}):
         self.model_file = model_kwargs.pop('model_file', None)
         self.model = self.model_loader(*model_kwargs)
 
+    @property
+    @abstractmethod
+    def model_uri(self):
+        pass
+
+    @model_uri.setter
+    def model_uri(self,val):
+        self.model_uri=val 
+        
```

Comment:
i meant to say: model_file should be an object property right? we actually need it to be persisted / inspectable, so we should have that specified as  property

Reply:
even if it'll somtimes be NONE

Reply:
yeah. hadn't thought about that. making a quick fix

Reply:
it still makes sense to have this be a child class of base onsite llm right?